src/storage/secondary/column/concrete_column_iterator.rs

```diff
@@ -275,29 +191,17 @@ impl<A: Array, F: BlockIteratorFactory<A>> ConcreteColumnIterator<A, F> {
         }
         assert_eq!(cnt, 0);
 
-        self.is_fake_iter = true;
-        self.block_iterator = self.factory.get_fake_iterator(
-            self.column.index().index(self.current_block_id),
-            self.current_row_id as usize,
-        )
+        // indicate that a new block (located by `current_block_id`) should be
+        // loaded at the beginning of the next `next_batch`
+        self.block_changed = true;
```

Comment:
If `skip` is called consequently for two times, things would go wrong. On L168, we will never be able to get the correct remaining_items from the block. `let remaining_items = self.block_iterator.remaining_items();`

Personally, I would still prefer retaining the `FakeBlockIterator`, so that the logic could be simple for all block kinds.

Reply:
Please also add unit tests for the `skip` function?

Reply:
updated